test/EndToEnd/tests/PackageRestoreTest.ps1

```diff
@@ -330,6 +330,31 @@ function Test-PackageRestore-AllSourcesAreUsed {
     }
 }
 
+# Test unneeded file are not created during restore
+function Test-PackageRestore-NoContent_TypesXml {
```

Comment:
Add a unit test instead of a functional test, and add a test for a package that has that file not at the root (make sure it is still extracted)